iotdb-core/confignode/src/test/java/org/apache/iotdb/confignode/manager/load/balancer/region/DestNodeSelectorTest.java

```diff
@@ -0,0 +1,131 @@
+
+  @BeforeClass
+  public static void setUp() {
+    // Construct TEST_DATA_NODE_NUM DataNodes
+    for (int i = 1; i <= TEST_DATA_NODE_NUM; i++) {
+      AVAILABLE_DATA_NODE_MAP.put(
+          i, new TDataNodeConfiguration().setLocation(new TDataNodeLocation().setDataNodeId(i)));
+      FREE_SPACE_MAP.put(i, Math.random());
+    }
+  }
+
+  @Test
+  public void testSelectDestNode() {
+    final int dataRegionGroupNum =
+        DATA_REGION_PER_DATA_NODE * TEST_DATA_NODE_NUM / DATA_REPLICATION_FACTOR;
+
+    List<TRegionReplicaSet> allocateResult = new ArrayList<>();
+    for (int index = 0; index < dataRegionGroupNum; index++) {
+      allocateResult.add(
+          ALLOCATOR.generateOptimalRegionReplicasDistribution(
+              AVAILABLE_DATA_NODE_MAP,
+              FREE_SPACE_MAP,
+              allocateResult,
+              allocateResult,
+              DATA_REPLICATION_FACTOR,
+              new TConsensusGroupId(TConsensusGroupType.DataRegion, index)));
+    }
+    List<TRegionReplicaSet> migratedReplicas =
+        allocateResult.stream()
+            .filter(
+                replicaSet -> replicaSet.getDataNodeLocations().contains(REMOVE_DATANODE_LOCATION))
+            .collect(Collectors.toList());
+
+    AVAILABLE_DATA_NODE_MAP.remove(REMOVE_DATANODE_LOCATION.getDataNodeId());
+    FREE_SPACE_MAP.remove(REMOVE_DATANODE_LOCATION.getDataNodeId());
+
+    List<TRegionReplicaSet> remainReplicas = new ArrayList<>();
+
+    for (TRegionReplicaSet replicaSet : migratedReplicas) {
+      List<TDataNodeLocation> dataNodeLocations = replicaSet.getDataNodeLocations();
+      allocateResult.remove(replicaSet);
+      dataNodeLocations.remove(REMOVE_DATANODE_LOCATION);
+      replicaSet.setDataNodeLocations(dataNodeLocations);
+      allocateResult.add(replicaSet);
+      remainReplicas.add(replicaSet);
+    }
+
+    Set<Integer> selectedNodeIds = new HashSet<>();
+
+    for (TRegionReplicaSet remainReplicaSet : remainReplicas) {
+      TDataNodeConfiguration selectedNode =
+          SELECTOR.selectDestDataNode(
+              AVAILABLE_DATA_NODE_MAP,
+              FREE_SPACE_MAP,
+              allocateResult,
+              allocateResult,
+              DATA_REPLICATION_FACTOR,
+              remainReplicaSet.regionId,
+              remainReplicaSet);
+      LOGGER.info(
+          "Selected DataNode {} for Region {}",
+          selectedNode.getLocation().getDataNodeId(),
+          remainReplicaSet.regionId);
+      allocateResult.remove(remainReplicaSet);
+      List<TDataNodeLocation> dataNodeLocations = remainReplicaSet.getDataNodeLocations();
+      dataNodeLocations.add(selectedNode.getLocation());
+      remainReplicaSet.setDataNodeLocations(dataNodeLocations);
+      allocateResult.add(remainReplicaSet);
+      selectedNodeIds.add(selectedNode.getLocation().getDataNodeId());
+    }
+
+    Assert.assertEquals(TEST_DATA_NODE_NUM - 1, selectedNodeIds.size());
```

Comment:
Should we calculate how many new regions are added to each node, whether they are balanced, and so on? Otherwise this is a little bit broad at the moment？

maybe something like this
<img width="1066" alt="image" src="https://github.com/user-attachments/assets/b9459f85-b4e6-41d3-a7e3-7482fa24fb2e" />


<img width="1062" alt="image" src="https://github.com/user-attachments/assets/1ebbbcb5-b6ce-4b30-9b8c-79976d7739fd" />

Reply:
added.